### ISSUE TITLE
OverflowError: writing pandas data frame to hdf5

### ISSUE THREAD

**Issue by DavidTolan** [PyTables/PyTables#531]
sorry I can't give you more details, but much of the data is confidential

**Versions of code**
pandas                    0.17.0              np110py27_0
pytables                  3.2.2               np110py27_1

shape of the dataframe (it's quite big)
(57163046, 24)

I have printed out the min/max of all int64s in the frame just before storing, eg
print frame[frame.dtypes[frame.dtypes == "int64"].index.tolist()].max()
none of the max are close to 2^63, nor are there any large negative numbers.

some of the data frame fields are strings, perhaps 80 chars or longer.

**original call to pandas**
frame.to_hdf(hdf5_file_name, self.key_name(name_tuple), **complevel=9, complib='blosc')**

trace back
  File "C:\Anaconda2\lib\site-packages\log_file_parser-0.1.10-py2.7.egg\log_file_parser\src\hdf5.py", line 39, in write_frames
    frame.to_hdf(hdf5_file_name, self.key_name(name_tuple), complevel=9, complib='blosc')
  File "C:\Anaconda2\lib\site-packages\pandas\core\generic.py", line 938, in to_hdf
    return pytables.to_hdf(path_or_buf, key, self, **kwargs)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 270, in to_hdf
    f(store)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 265, in <lambda>
    f = lambda store: store.put(key, value, **kwargs)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 827, in put
    self._write_to_group(key, value, append=append, **kwargs)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 1265, in _write_to_group
    s.write(obj=value, append=append, complib=complib, **kwargs)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 2805, in write
    self.write_array('block%d_values' % i, blk.values, items=blk_items)
  File "C:\Anaconda2\lib\site-packages\pandas\io\pytables.py", line 2572, in write_array
    vlarr.append(value)
  File "C:\Anaconda2\lib\site-packages\tables\vlarray.py", line 537, in append
    self._append(nparr, nobjects)
  File "tables\hdf5extension.pyx", line 1929, in tables.hdf5extension.VLArray._append (tables\hdf5extension.c:20849)
OverflowError: Python int too large to convert to C long

I have a work around by discarding some less important data, and ending up with a much smaller dataframe. It looks like the problem is with the size of the frame, but it could also be in the discarded data.

If you have any specific questions, on reproducing this, I will help as best I can.

**Comment by DavidTolan** [comment]
The problem also occurs in 

# packages in environment at C:\Anaconda:
#
pandas                    0.17.1              np110py27_0

**Comment by andreabedini** [comment]
[This](https://github.com/PyTables/PyTables/blob/v.3.2.2/tables/hdf5extension.pyx#L1929) is the incriminated line. `_append` is declared `def _append(self, ndarray nparr, int nobjects)`, and this breaks if nobjects doesn't fit into an C sized int.

**Comment by MInner** [comment]
@andreabedini yes, I also experience same error on the same line, when trying to store a large list of arrays; don't you know if there's a fix coming? `tables==3.2.2`.

I'm actually using `deepdish` warper.

**Comment by FrancescAlted** [comment]
I had a quick try at this in PR #550, but the test included still segfaults (see details in PR).  I don't have time right now to continue debugging, so if any other can volunteer some time on it that would be great.

**Comment by Horta** [comment]
Still having this problem on '0.18.1'

**Comment by gregroberts** [comment]
Just found this issue, I'm on pandas 0.20.1 && PyTables 3.2.2. 

Trying to put a df into hdf:

df.shape:
(12359531, 11)

df.dtypes:

user_id                     int64
date               datetime64[ns]
session_uuid               object
pagepath                   object
source_medium              object
campaign                   object
device_category            object
pageviews                   int64
dd                         object
source                     object
medium                     object
dtype: object

**Comment by omor1** [comment]
I seem to be having this same problem. I think I'm generating a huge (larger than 4 GiB) Pandas DataFrame and this is causing issues, specifically when using the `put` method, causing an `OverflowError` in `tables.hdf5extension.VLArray._append`. The current method creates a dense DataFrame, but with many NaN/None values; I'll see if creating a sparse DataFrame helps my particular use case.
